pdns/base64.cc

```diff
@@ -58,8 +62,10 @@ std::string Base64Encode (const std::string& src)
     bio = BIO_new(BIO_s_mem());
     bio = BIO_push(b64, bio);
     BIO_set_flags(bio, BIO_FLAGS_BASE64_NO_NL);
-    BIO_write(bio, src.c_str(), src.length());
-    BIO_flush(bio);
+    if (BIO_write(bio, src.c_str(), src.length() != src.length())) {
```

Comment:
I think you have misplaced a ')', here src.length() != src.length() is not going to work well :)